### PR TITLE
Update SDK and custom converter/decorator samples

### LINKED ISSUE #14: Sample request: Custom decorator
Inspired by recent Slack convo, something like an `auto_heartbeater`:

```python
@activity.defn
@activity_utils.auto_heartbeater(interval=0.5)
async def my_activity():
    ...
```

### LINKED ISSUE #15: Sample request: Custom converter
Need to show how to support a custom type w/ a custom converter

### PR DESCRIPTION
**Pull request by cretz**
## What was changed

* Update SDK
* Custom payload converter sample
* Custom decorator sample

## Checklist

1. Closes #15
2. Closes #14

### CODE REVIEW

File: custom_converter/worker.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+class GreetingWorkflow:
+    @workflow.run
+    async def run(self, input: GreetingInput) -> GreetingOutput:
+        return GreetingOutput(f"Hello, {input.name}")
+
+
+class GreetingEncodingPayloadConverter(EncodingPayloadConverter):
+    @property
+    def encoding(self) -> str:
+        return "my-greeting-encoding"
```
Comment (bergundy) [inline custom_converter/worker.py:37]: Maybe use a "namespace" here like `text/greeting` to encourage this practice.

File: custom_decorator/activity_utils.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+def auto_heartbeater(fn: F) -> F:
+    # We want to ensure that the type hints from the original callable are
+    # available via our wrapper, so we use the functools wraps decorator
+    @wraps(fn)
+    async def wrapper(*args, **kwargs):
+        # Heartbeat twice as often as the timeout
+        heartbeat_timeout = activity.info().heartbeat_timeout
+        if heartbeat_timeout:
+            asyncio.create_task(heartbeat_every(heartbeat_timeout.total_seconds() / 2))
```
Comment (bergundy) [inline custom_decorator/activity_utils.py:19]: When would this stop heartbeating?

File: custom_decorator/activity_utils.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return await fn(*args, **kwargs)
+
+    return cast(F, wrapper)
+
+
+async def heartbeat_every(delay: float, *details: Any) -> None:
+    # Heartbeat every so often while not cancelled
+    while not activity.is_cancelled():
```
Comment (bergundy) [inline custom_decorator/activity_utils.py:27]: Does `is_cancelled` return `True` if activity completes?

File: custom_decorator/activity_utils.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+from typing import Any, Awaitable, Callable, TypeVar, cast
+
+from temporalio import activity
+
+F = TypeVar("F", bound=Callable[..., Awaitable[Any]])
+
+
+def auto_heartbeater(fn: F) -> F:
+    # We want to ensure that the type hints from the original callable are
+    # available via our wrapper, so we use the functools wraps decorator
```
Comment (bergundy) [inline custom_decorator/activity_utils.py:13]: Generally good practice for any decorator

**Review by bergundy — APPROVED** [review]

File: custom_converter/worker.py (hunk 1 above)
Comment (cretz) [reply custom_converter/worker.py:37]: Ok

File: custom_decorator/activity_utils.py (hunk 2 above)
Comment (cretz) [reply custom_decorator/activity_utils.py:19]: Only on cancel which is fine for this demonstration but not for general purpose use. Per comment below, I will fix.

Comment (cretz) [reply custom_decorator/activity_utils.py]: Fixed

File: custom_converter/worker.py (hunk 1 above)
Comment (cretz) [reply custom_converter/worker.py]: Fixed

File: custom_decorator/activity_utils.py (hunk 3 above)
Comment (cretz) [reply custom_decorator/activity_utils.py]: Fixed

**Pull request merged**